Sources/TuistCache/Extensions/GraphTraverser+Extras.swift

```diff
@@ -0,0 +1,36 @@
+import TuistCore
+import TuistGraph
+
+extension GraphTraversing {
+    /// Returns the included based on the parameters.
+    public func filterIncludedTargets<GraphTargets: Collection>(
+        basedOn targets: GraphTargets,
+        testPlan: String?,
+        includedTargets: Set<String>,
+        excludedTargets: Set<String>,
+        excludingExternalTargets: Bool = false
+    ) -> Set<GraphTarget> where GraphTargets.Element == GraphTarget {
+        let allTestPlansTargetNames: Set<String>?
+        if includedTargets.isEmpty, let testPlanName = testPlan, let testPlan = self.testPlan(name: testPlanName) {
+            allTestPlansTargetNames = Set(testPlan.testTargets.filter(\.isEnabled).map(\.target.name))
+        } else {
+            allTestPlansTargetNames = nil
+        }
+
+        lazy var allInternalTargets = allInternalTargets().map(\.target.name)
```

Comment:
Why is this var `lazy`?

Reply:
It is lazy as it is not used in all code path, so it's initialized only when used.